inference-engine/thirdparty/clDNN/tests/test_cases/gather_nd_gpu_test.cpp

```diff
@@ -0,0 +1,730 @@
+// Copyright (c) 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#include <gtest/gtest.h>
+
+#include <api/input_layout.hpp>
+#include <api/memory.hpp>
+#include <api/gather_nd.hpp>
+#include <api/topology.hpp>
+#include <api/network.hpp>
+
+#include <cstddef>
+#include <tests/test_utils/test_utils.h>
+
+using namespace cldnn;
+using namespace ::tests;
+
+inline void DoTest(const engine& engine,
+    const cldnn::memory& input0,
+    const cldnn::memory& input1,
+    const std::vector<float>& expected_results,
+    const int indices_rank,
+    const int batch_dims) {
+    topology topology;
+    topology.add(input_layout("InputData", input0.get_layout()));
+    topology.add(input_layout("InputIndices", input1.get_layout()));
+    topology.add(
+        gather_nd("gather_nd", "InputData", "InputIndices", indices_rank, batch_dims)
+    );
+
+    network network(engine, topology);
+
+    network.set_input_data("InputData", input0);
+    network.set_input_data("InputIndices", input1);
+    auto outputs = network.execute();
+    auto output = outputs.at("gather_nd").get_memory();
+    auto output_ptr = output.pointer<uint16_t>();
+
+    for (size_t i = 0; i < expected_results.size(); ++i) {
+        EXPECT_EQ(expected_results[i], float16_to_float32(output_ptr[i]));
+    }
+}
+
+TEST(gather_nd_gpu_fp16, d23322_i231312_ir6_batch2) {
+    engine engine;
```

Comment:
Please use statically created test engine (`get_test_engine()`) to reduce tests execution time.

Reply:
I applied it.